### PR TITLE
JS implementation of Fibonacci search added

### PR DESCRIPTION
**Pull request by nipunbharti**
**Fixes issue:** #3048 

**Changes:**
JS implementation added

### CODE REVIEW

File: code/search/fibonacci_search/fibonacci_search.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+	if(firstPreceding && ar[range+1] == x){
+		return range + 1;
+	}
+
+	return -1;
+
+}
+function min(a,b) {
+	if(a > b){
```
Comment (arnavb) [inline code/search/fibonacci_search/fibonacci_search.js]: Use the ternary operator here.

File: code/search/fibonacci_search/fibonacci_search.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
+function fibonacciSearch(ar,x,n) {
+	var secondPreceding = 0,
+	firstPreceding = 1;
```
Comment (arnavb) [inline code/search/fibonacci_search/fibonacci_search.js]: Any reason why you aren’t using var here?

Comment (nipunbharti) [reply code/search/fibonacci_search/fibonacci_search.js]: There is a comma to separate the two.

File: code/search/fibonacci_search/fibonacci_search.js (hunk 1 above)
Comment (nipunbharti) [reply code/search/fibonacci_search/fibonacci_search.js]: Okay I'll fix it

File: code/search/fibonacci_search/fibonacci_search.js (hunk 2 above)
Comment (arnavb) [reply code/search/fibonacci_search/fibonacci_search.js]: Oh right, didn’t see that. I think you should put both variables on the same line.

File: code/search/fibonacci_search/fibonacci_search.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+	if(firstPreceding && ar[range+1] == x){
+		return range + 1;
+	}
+
+	return -1;
+
+}
+function min(a,b) {
+	a > b ? return b : return a;
```
Comment (arnavb) [inline code/search/fibonacci_search/fibonacci_search.js]: return a > b ? b : a;

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**